Mostly Fixed Packaging

slaymaker1907 opened the pull request:
There are still some mysterious bugs with some of the plugins where you can't create edges for some reason (the circuit plugin seems to work, though).

A couple of changes I also implemented was to give process.env.ENV a less confusing name of sinap.env. I also removed the boundary for the new file window which allows for developer tools and a clean menu.

slaymaker1907 commented:
See #217 for the remaining issue.

ilargierdi commented:
I can't open anything on a fresh clone on CentOS. The console is saying that it can't find the plugin associated with the file. I tried going through the new file dialog box, but nothing would load there, either.

slaymaker1907 commented:
@ilargierdi make sure that you are using npm version >= 4.2 (the lowest I've gotten to work). I was able to reproduce your error on Ubuntu, but it turned out I was using npm 3 instead of npm 4. Once I updated npm via

```bash
sudo npm install -g npm
````

it now seems to work. I also added a note in the README.md about this dependency.

ilargierdi commented:
I'm still getting the error. I'm running on npm 4.4.1 and node 6.9.4.

slaymaker1907 commented:
I will install a CentOS VM and see if I can replicate this.

slaymaker1907 commented:
I was able to get everything working on a CentOS 7 VM. I can send you a copy of the VM if you would like to try it out and compare it to your setup. For some reason it wouldn't build Windows binaries, but I think that may just have been particular to my setup and not Sinap's fault.

ilargierdi commented:
If you're talking about running from the dist folder after `npm package`, that works fine. I'm talking about `npm start`; I can't open anything or even create a new file if running from the build directory. Also, I'm not sure if I like the borderless dialog window. Mostly because I miss the close button at the top.

slaymaker1907 commented:
@ilargierdi I think your issue was that you did not run a regular build after running package. The production build is slightly different than the development build, in particular regarding where plugins are loaded.

I just pushed a small change so that it will remove the build folder after packaging to try and help avoid these issues.

build:dll is the same in both production and in development so I don't delete that folder.

ilargierdi commented:
The bug I'm experiencing was happening before I event attempted to `npm run package:linux`. I hadn't even used that command until my last comment. From a fresh install, build:dll, and build, if I then try `npm start`, the application runs fine, but I can't open anything or even create a new file.

slaymaker1907 commented:
Ok, I think I fixed the issue. I misunderstood the electron app.md documentation.

ilargierdi commented:
It works now!

slaymaker1907 commented:
I implemented the suggestion of @RighteousRaichu to make the new file window bordered as well as adding in a debug flag separate from the production flag.

slaymaker1907 commented:
@Sheyne or @RighteousRaichu can you try this out on MacOS?

RighteousRaichu commented on app/models/menu.ts:
```diff
@@ -242,16 +242,11 @@ if (process.platform === 'darwin') {
-            role: 'toggledevtools',
-        });
-    }
+let viewSubmenu = viewMenu.submenu;
```
I'm gonna add a check here for debug mode. Any objections @slaymaker1907

RighteousRaichu replied:
`(node:75895) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: ENOENT: no such file or directory, open '/Users/Danie
l/Development/sinap-ide/build/package.json'`

I'm gonna look into this but any ideas @slaymaker1907 ?

RighteousRaichu replied:
I think we should address #167 before merging this in. Currently from a packaged build I can't open examples that have a custom image because it can't find the image file.

RighteousRaichu commented:
Some things we need before we can merge this:
Figure out why on my computer I can't package twice in a row without cleaning first.
Find workaround for missing typescript definition files.
Expose plugin directory to Plugin so custom images work.

slaymaker1907 commented:
I haven't figured out what's been going on for @RighteousRaichu with duplicate package. However, typescript lib files are now bundled in the app and the plugin directory is exposed to the graph editor (kind of a hack for now until we do plugin specific resources which requires stuff from core).

With these changes, we should be able to ship the app to third party users.

RighteousRaichu commented:
I can't reproduce the problem anymore and everything looks great rn. Feel free to merge this in when you're ready @slaymaker1907
